### ISSUE TITLE
tenants stuck in "Waiting for pod get ready"

### ISSUE THREAD

**Issue by ElisaMeng** [minio/operator#556]
Running helm deployment via following the instructions, tennats minio1 failed to get ready. 

```
kubectl -n minio logs -f pod/minio1-ss-0-0

API: SYSTEM()
Time: 16:23:21 UTC 03/25/2021
Error: lookup minio1-ss-0-2.minio1-hl.minio.svc. on 10.43.0.10:53: no such host (*net.DNSError)
       host=minio1-ss-0-2.picloud-hl.minio.svc., elapsedTime=11 minutes elapsed
       5: cmd/endpoint.go:469:cmd.Endpoints.UpdateIsLocal()
       4: cmd/endpoint.go:653:cmd.CreateEndpoints()
       3: cmd/endpoint-ellipses.go:372:cmd.createServerEndpoints()
       2: cmd/server-main.go:142:cmd.serverHandleCmdArgs()
       1: cmd/server-main.go:418:cmd.serverMain()

API: SYSTEM()
Time: 16:23:21 UTC 03/25/2021
Error: lookup minio1-ss-0-3.minio1-hl.minio.svc. on 10.43.0.10:53: no such host (*net.DNSError)
^C       elapsedTime=11 minutes elapsed, host=minio1-ss-0-3.minio1-hl.minio.svc.
       5: cmd/endpoint.go:469:cmd.Endpoints.UpdateIsLocal()

```
While trying manually, dns name is actually resolvable

```
[root@ minio1-ss-0-0 /]# curl -v minio1-ss-0-2. minio1-hl.minio.svc.
* Rebuilt URL to: minio1-ss-0-2. minio1-hl.minio.svc./
*   Trying 10.42.2.189...
* TCP_NODELAY set
* connect to 10.42.2.189 port 80 failed: Connection refused
* Failed to connect to minio1-ss-0-2. minio1-hl.minio.svc port 80: Connection refused
* Closing connection 0
curl: (7) Failed to connect to minio1-ss-0-2. minio1-hl.minio.svc port 80: Connection refused

```

## Expected Behavior

Work out of the box 

## Current Behavior

Failed out of the box 

## Steps to Reproduce (for bugs)

1. helm install --namespace minio-operator --create-namespace --generate-name minio/minio-operator
2. Wait for a while to access console
3. In console, the tenants minio1 stuch in "Waiting for pod get ready", tenant is not usable. 
4.

## Your Environment

* Version used (`minio-operator`): 4.0.3
* Environment name and version (e.g. kubernetes v1.17.2): v1.19.8
* Server type and version: baremetal
* Operating System and version (`uname -a`): ubuntu  linux 5.4.0
* Link to your deployment file:

**Comment by dvaldivia** [comment]
hello @ElisaMeng, there seems to be a space in your tenant name `minio1-ss-0-2. minio1-hl.minio.svc`, perhaps we should trim the spaces in helm chart values @nitisht @harshavardhana 

Can you try to re-attempt the deployment making sure there's no spaces at the beginign of the tenant name @ElisaMeng ?

**Comment by Sleepy-GH** [comment]
I ran into the same issue just now too and figured out that there's no default for the operator.clusterDomain value. Set this value to your cluster domain (default cluster.local) and it works.

**Comment by dvaldivia** [comment]
that makes sense @Sleepy-GH , we need to patch that

**Comment by alexkorotysh** [comment]
We have the same issue, I've tried to set env in helm, but looks like the helm chart could not work with env in the tenant section of the YAML file

**Comment by ElisaMeng** [comment]
@dvaldivia Sorry, there was not a space. it is just my typo. 
@Sleepy-GH  I take your tips, it move further,  get stuck  at "Provisioning  console deployment "

and one of the pods output like this:

```
Status:         4 Online, 0 Offline. 
Endpoint: https://10.42.0.49:9000  https://127.0.0.1:9000     

Browser Access:
   https://10.42.0.49:9000  https://127.0.0.1:9000    

Object API (Amazon S3 compatible):
   Go:         https://docs.min.io/docs/golang-client-quickstart-guide
   Java:       https://docs.min.io/docs/java-client-quickstart-guide
   Python:     https://docs.min.io/docs/python-client-quickstart-guide
   JavaScript: https://docs.min.io/docs/javascript-client-quickstart-guide
   .NET:       https://docs.min.io/docs/dotnet-client-quickstart-guide
Waiting for all MinIO IAM sub-system to be initialized.. lock acquired
2021-03-25 21:21:19.328599 I | http: TLS handshake error from 10.42.2.199:58084: remote error: tls: bad certificate

```

what else is missing here?

**Comment by Sleepy-GH** [comment]
Hard to say without knowing what values you've used with your helm deployment and your tenant spec

**Comment by ElisaMeng** [comment]
@Sleepy-GH  I did not change anything except disk number, just follow the helm instructions

**Comment by Fabyao** [comment]
@ElisaMeng I am getting the same TLS handshake errors and `User is not allowed to be same as admin access key`. Did you resolve it?

**Comment by mnp** [comment]
@Sleepy-GH @dvaldivia 
> I ran into the same issue just now too and figured out that there's no default for the operator.clusterDomain value. Set this value to your cluster domain (default cluster.local) and it works.

Hey would you please be more specific about where this should be set? I might be hitting this issue also.

**Comment by Sleepy-GH** [comment]
> @Sleepy-GH @dvaldivia
> 
> > I ran into the same issue just now too and figured out that there's no default for the operator.clusterDomain value. Set this value to your cluster domain (default cluster.local) and it works.
> 
> Hey would you please be more specific about where this should be set? I might be hitting this issue also.

@mnp When you install the operator using Helm you can either set it as value. The default for most Kubernetes cluster is cluster.local, so set it to that value and you should be good (unless you're using a custom domain for you cluster)

**Comment by mnp** [comment]
@Sleepy-GH - Thank you, that worked as expected, for data point on GKE.
```
helm install -n blue --set operator.clusterDomain=cluster.local r1 helm-releases/minio-operator-4.0.7-1.tgz
```

**Comment by harshavardhana** [comment]
Released 4.0.8 - please test and let us know if you see the same problem. 

This PR should fix this

```
commit 64f7b32429873e49f6b3a314405ad70aee6e3f9e
Author: Jay Li <xenium_lee@163.com>
Date:   Mon Mar 29 16:08:00 2021 +0800

    Fix: Empty env should not be included (#559)
    
    os.LookupEnv()
    LookupEnv retrieves the value of the environment variable
    named by the key. If the variable is present in the environment
    the value (which may be empty) is returned and the boolean is true.

diff --git a/helm/minio-operator/templates/operator-deployment.yaml b/helm/minio-operator/templates/operator-deployment.yaml
index 6d82605..a7593f5 100644
--- a/helm/minio-operator/templates/operator-deployment.yaml
+++ b/helm/minio-operator/templates/operator-deployment.yaml
@@ -24,11 +24,17 @@ spec:
         - name: {{ .Chart.Name }}
           image: "{{ .Values.operator.image.repository }}:{{ .Values.operator.image.tag }}"
           imagePullPolicy: {{ .Values.operator.image.pullPolicy }}
+          {{- if or .Values.operator.clusterDomain .Values.operator.nsToWatch }}
           env:
-          - name: CLUSTER_DOMAIN
-            value: {{ .Values.operator.clusterDomain }}
-          - name: WATCHED_NAMESPACE
-            value: {{ .Values.operator.nsToWatch }}
+            {{- if .Values.operator.clusterDomain }}
+            - name: CLUSTER_DOMAIN
+              value: {{ .Values.operator.clusterDomain }}
+            {{- end }}
+            {{- if .Values.operator.nsToWatch }}
+            - name: WATCHED_NAMESPACE
+              value: {{ .Values.operator.nsToWatch }}
+            {{- end }}
+          {{- end }}
           resources:
             {{- toYaml .Values.operator.resources | nindent 12 }}
           securityContext:
```
